File: georocket-server/src/test/java/io/georocket/NetUtils.java
package io.georocket;

import java.io.IOException;
import java.net.ServerSocket;

/**
 * Helper class for server tests.
 * @author Benedikt Hiemenz
 */
public class NetUtils {

  /**
   * Find a free socket port.
   * @return the number of the free port
   */
  public static int findPort() {
    try (ServerSocket socket = new ServerSocket(0)) {
      return socket.getLocalPort();
    } catch (IOException e) {
      throw new RuntimeException("Could not find a free port");
    }
  }
}

File: kotlin_impl/src/test/kotlin/io/georocket/NetUtils.kt
package io.georocket

import java.net.ServerSocket
import java.io.IOException
import java.lang.RuntimeException

/**
 * Helper class for server tests.
 * @author Benedikt Hiemenz
 */
object NetUtils {
    /**
     * Find a free socket port.
     * @return the number of the free port
     */
    fun findPort(): Int {
        try {
            ServerSocket(0).use { socket -> return socket.localPort }
        } catch (e: IOException) {
            throw RuntimeException("Could not find a free port")
        }
    }
}
